consensus/types/src/beacon_response.rs

```diff
@@ -0,0 +1,239 @@
+use crate::{ContextDeserialize, ForkName};
+use serde::de::DeserializeOwned;
+use serde::{Deserialize, Deserializer, Serialize};
+use serde_json::value::Value;
+
+pub trait ForkVersionDecode: Sized {
+    /// SSZ decode with explicit fork variant.
+    fn from_ssz_bytes_by_fork(bytes: &[u8], fork_name: ForkName) -> Result<Self, ssz::DecodeError>;
+}
+
+/// The metadata of type M should be set to `EmptyMetadata` if you don't care about adding fields other than
+/// version. If you *do* care about adding other fields you can mix in any type that implements
+/// `Deserialize`.
+#[derive(Debug, PartialEq, Clone, Serialize)]
+pub struct ForkVersionedResponse<T, M = EmptyMetadata> {
+    pub version: ForkName,
+    #[serde(flatten)]
+    pub metadata: M,
+    pub data: T,
+}
+
+// Used for responses to V1 endpoints that don't have a version field.
+/// The metadata of type M should be set to `EmptyMetadata` if you don't care about adding fields other than
+/// version. If you *do* care about adding other fields you can mix in any type that implements
+/// `Deserialize`.
+#[derive(Debug, PartialEq, Clone, Serialize)]
+pub struct UnVersionedResponse<T, M = EmptyMetadata> {
```

Comment:
Personally I think this looks nicer? "unversioned" is a valid word on its own

```suggestion
pub struct UnversionedResponse<T, M = EmptyMetadata> {
```

Reply:
I've changed this in:

- 2ce2b28b1b455aa3378ad4cea4c2fe754bedb968

Lemme know if you prefer to change it back.

Reply:
LGTM!